src/storage/leveldb/nanocontract_index.ts

```diff
@@ -0,0 +1,118 @@
+import path from 'path';
+import { Level } from 'level';
+import { AbstractSublevel } from 'abstract-level';
+import { IKVNanoContractIndex, INcData } from "src/types";
+import { errorCodeOrNull, KEY_NOT_FOUND_CODE } from './errors';
+import { checkLevelDbVersion } from '../../utils/storage';
+
+export const REGISTERED_PREFIX = 'registered';
+
+export default class LevelNanoContractIndex implements IKVNanoContractIndex {
+  dbpath: string;
+  /**
+   * Registered Nano Contract database
+   * Key: nckey
+   * Value: INcData (json encoded)
+   */
+  registeredDB: AbstractSublevel<Level, string | Buffer | Uint8Array, string, INcData>;
+  indexVersion: string = '0.0.1';
+
+  constructor(dbpath: string) {
+    this.dbpath = path.join(dbpath, 'nanocontract');
+    const db = new Level(this.dbpath);
+    this.registeredDB = db.sublevel<string, INcData>(REGISTERED_PREFIX, { valueEncoding: 'json' });
+  }
+
+  /**
+   * Close the database and the sublevel children.
+   * @returns {Promise<void>}
+   */
+  async close(): Promise<void> {
+    await this.registeredDB.db.close();
+  }
+
+  /**
+   * Check that the index version matches the expected version.
+   * @returns {Promise<void>}
+   */
+  async checkVersion(): Promise<void> {
+    const db = this.registeredDB.db;
+    const instanceName = this.constructor.name;
+    await checkLevelDbVersion(instanceName, db, this.indexVersion);
+  }
+
+  /**
+   * Validate the database.
+   * @returns {Promise<void>}
+   */
+  async validate(): Promise<void> {
+    await this.checkVersion();
+  }
+
+  /**
+   * Delete all entries on the database.
+   */
+  async clear(): Promise<void> {
+      await this.registeredDB.db.clear();
```

Comment:
Is this indentation correct?

Reply:
Fixed at https://github.com/HathorNetwork/hathor-wallet-lib/pull/614/commits/f7f228e35c65a12e4ab9eade924898eb8c63348b